File: Data Structures & Algorithms/search-2d-matrix/submission-1.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) { 
        int right = matrix.size() * matrix[0].size() - 1;
        int left = 0;
        while(left <= right){

        int mid = (right + left)/2;
        int row = mid/matrix[0].size(); // the row
        int col = mid % matrix[0].size() ; // the column
        if(matrix[row][col] == target){
            return true;
        }
        if(matrix[row][col] > target){
            right = mid-1; 
        }
        if(matrix[row][col] < target){
            left = mid+1;
        }

        }
        return false;

    }
};

File: Data Structures & Algorithms/set-zeroes-in-matrix/submission-1.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        r, c = len(matrix), len(matrix[0])
        zero = []
        for i in range(r):
            for j in range(c):
                if matrix[i][j] == 0:
                    zero.append([i,j])
        for z in zero:
            for i in range(r):
                matrix[i][z[1]] = 0
            for j in range(c):
                matrix[z[0]][j] = 0
        
